docs/LemMinX-Extensions.md

```diff
@@ -101,6 +101,39 @@ The [LemMinx Extensions API](https://github.com/eclipse/lemminx/tree/master/org.
 - Formatter with [IFormatterParticipant](https://github.com/eclipse/lemminx/blob/master/org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/services/extensions/format/IFormatterParticipant.java)
 - Symbols with [ISymbolsProviderParticipant](https://github.com/eclipse/lemminx/blob/master/org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/services/extensions/ISymbolsProviderParticipant.java)
 
+## XML Language Server services available for extensions
+XML Language Server extension may need to use standard Language Server features such as commands, documents and ability to manipulate documents. These are available to extensions indirectly via specialized service API.
+
+#### Command Service
+The command service is a wrapper around Language Server client functionality allowing to register/unregister/execute commands via Language Server Protocol.
```

Comment:
I have several comments about your documentation:

1. IMHO I think the sample is little confused, because it mixes client command to execute by the server and the command server.

At first we should explain that there are 2 kind of command:

 * a client command that we can execute from the server (I mean executeClientCommand)
 * a server command that we can execute from the client (command are registered on server side with a unique command id). 

Please split your Command Service in 2 parts to explain that by given a sample.

2. The second comments is that the sample doesn't explain how to get the commandService.

3. please use markdow syntax to add java code sample:

```java
// Command registration
commandService.registerCommand("my-cmd", (params, cancelChecker) -> "executed");
```

Thanks!

Reply:
Thanks for making a note of it. Corrected now